server/__tests__/loginAPI.test.js

```diff
@@ -0,0 +1,99 @@
+import { expect } from 'chai';
+import request from 'supertest';
+import { getMockUser } from '../src/utils/testUtils';
+// import User from '../src/database/models/user';
+
+let mockUser;
+
+const { testServer } = global;
+
+beforeAll(async (done) => {
+  mockUser = getMockUser();
+  // Approach 1
+  // const user = new User(mockUser);
+  // await user.hashPassword();
+  // await user.save();
+  // done();
+  // Approach 2
+  await request(testServer)
+    .post('/api/signup')
+    .send(mockUser);
+  done();
+});
+
+// /api/login test suite
+describe('Login API', () => {
+  it('returns user on logging in with email and password', (done) => {
+    const { email, password } = mockUser;
+    request(testServer)
+      .post('/api/login')
+      .send({
+        email,
+        password,
+      })
+      .expect(200)
+      .end((error, response) => {
+        const createdUser = response.body;
+        expect(createdUser).to.have.keys('id', 'username');
```

Comment:
Shouldn't we check username also? to be sure that it's returning the username that we sent in the request.

Reply:
We are logging in using email right now. Logging in with username can be added later.